sdks/go/pkg/beam/io/bigqueryio/bigquery.go

```diff
@@ -208,29 +208,58 @@ func mustParseTable(table string) QualifiedTableName {
 // TODO(herohde) 7/14/2017: allow CreateDispositions and WriteDispositions. The default
 // is not quite what the Dataflow examples do.
 
+// WriteOptions represents additional options for executing a write
+type WriteOptions struct {
+	// CreateDisposition specifies the circumstances under which destination table will be created
+	CreateDisposition bigquery.TableCreateDisposition
+}
+
+// newWriteOptions creates a new instance of WriteOptions
+// "CreateIfNeeded" is set as the default write disposition
+func newWriteOptions() WriteOptions {
+	return WriteOptions{CreateDisposition: bigquery.CreateIfNeeded}
+}
+
+// WithCreateDisposition specifies the circumstances under which destination table will be created
+func WithCreateDisposition(cd bigquery.TableCreateDisposition) func(wo *WriteOptions) error {
```

Comment:
Instead of using a vanilla function, declare the function type *as* the exported options type.

`type WriteOption func(*writeOptions)`

This avoids the "you're using unexported types in an Exported API" style messages, and provides something we can document on.

This is preferable to simply having an aribitrary visible struct, when the intended mode isn't for users to write their own options functions. Having both "user can write this function" and "helper methods" it makes the API harder to use than "call functions X Y Z to set X Y Z"